feature-dapp-impl/src/main/java/io/novafoundation/nova/feature_dapp_impl/utils/tabs/models/PageSession.kt

```diff
@@ -0,0 +1,88 @@
+package io.novafoundation.nova.feature_dapp_impl.utils.tabs.models
+
+import android.content.Context
+import android.content.Intent
+import android.graphics.Bitmap
+import android.webkit.WebView
+import io.novafoundation.nova.feature_dapp_impl.web3.webview.PageCallback
+import io.novafoundation.nova.feature_dapp_impl.web3.webview.Web3ChromeClient
+import io.novafoundation.nova.feature_dapp_impl.web3.webview.Web3WebViewClient
+import io.novafoundation.nova.feature_dapp_impl.web3.webview.injectWeb3
+import io.novafoundation.nova.feature_dapp_impl.web3.webview.uninjectWeb3
+import java.util.Date
+
+class PageSessionFactory {
+
+    fun create(tabId: String, sessionStartTime: Date, startUrl: String): PageSession {
+        return PageSession(
+            tabId = tabId,
+            sessionStartTime = sessionStartTime,
```

Comment:
It seems like `sessionStartTime` could be computed by the factory itself
Actually this field wont be needed at all given we migrate to LruCache